paimon-core/src/main/java/org/apache/paimon/operation/FileDeletionBase.java

```diff
@@ -259,12 +267,17 @@ public void deleteAddedDataFiles(String manifestListName) {
 
     private void deleteAddedDataFiles(List<ExpireFileEntry> manifestEntries) {
         List<Path> dataFileToDelete = new ArrayList<>();
+        Map<Pair<BinaryRow, Integer>, DataFilePathFactory> dataFilePathFactoryMap = new HashMap<>();
```

Comment:
Extract a common util method?
These logic code has appeared multi times.

Reply:
I will do this.